Assets/Scripts/Core/Animation/AnimatorController.cs

```diff
@@ -0,0 +1,34 @@
+﻿using UnityEngine;
+
+namespace Core.Animation
+{
+    public abstract class AnimatorController : MonoBehaviour
+    {
+        private AnimationType _currentAnimationType;
+
+        public void PlayAnimation(AnimationType animationType, bool active)
```

Comment:
Imho, building abstaration on the fixed enum is too rigid. Esentially it means that all other entities will have the same set of animations